### DISCUSSION TITLE
Can someone explain the feature_engineering.py example?

### DISCUSSION THREAD

**Discussion by carljoseph** [PrefectHQ/prefect#4174] [Q&A] [answered]
[Here is the example file](https://github.com/PrefectHQ/prefect/blob/master/examples/feature_engineering.py#L274)

In particular, I'm interested to understand what line 274 is doing, and how is a dependency created between that line (which isn't a Prefect `Task`) and subsequent tasks?

When I do up my own version of this, any task that uses the output of `clean = DataFrame(...)` doesn't form a dependency on it.

**Comment by cicdw** [comment, marked as answer]
Hi @carljoseph - I'm not the author of that file, but I think the dependencies are mainly coming from the special `__getitem__` override on the `DataFrame` class here: https://github.com/PrefectHQ/prefect/blob/master/examples/feature_engineering.py#L232-L233

Note that `get` is defined above as a Prefect task, so all indexed calls like `clean["colname"]` on the `DataFrame` object result in new tasks and task dependencies.

**Reply by carljoseph** [reply]
Thanks for pointing that out to me. Really appreciate the assistance.
